ledger/src/bank_forks_utils.rs

```diff
@@ -251,14 +250,14 @@ fn bank_forks_from_snapshot(
                     .unwrap_or(0),
             );
             if bank.slot() < latest_snapshot_archive_slot {
-                error!(
-                    "Attempting to boot from local state at a slot {} that is *older* than the latest \
-                     snapshot archive slot {}, which is not supported. Either remove the snapshot archive \
-                     or remove the --boot-from-local-state CLI flag.",
+                warn!(
+                    "Starting up from local state at slot {}, which is *older* \
+                    than the latest snapshot archive at slot {}. If this is not \
+                    desired, remove the --ignore-snapshot-archives-at-startup \
+                    CLI flag and restart.",
                     bank.slot(),
                     latest_snapshot_archive_slot,
                 );
-                process::exit(1);
```

Comment:
Putting on a node operator hat, I think I'd want the snapshot archive to be used in this case. It's super intrusive to either delete the local state ("snapshot") or modify the command-line to remove `--ignore-snapshot-archives-at-startup` just this one time. The primary case I think people would run into this is when they download a new snapshot because their node fell behind for some reason, while generally preferring to use `--ignore-snapshot-archives-at-startup`.

That is, a `--prefer-local-state-at-startup` is I think what a node operator would be actually looking for. Also don't love "local state" but `--prefer-snapshot-at-startup` would be super confusing. Back to #terminology? 🫠

Reply:
I feel like taking a stab at the better name is worth it now so we don't need to re-litigate this discussion later. 

So maybe this?
  `--use-snapshot-archives-at-startup [always|never|newer] (default: always)`

"newer" gets added later and hopefully becomes the default - "OnlyWhenOlder" is "newer" in my example arg, since i can't see a reason why we'd want to use a snapshot archive that's older than the "local state"